src/main/java/com/mraof/minestuck/block/CassettePlayerBlock.java

```diff
@@ -0,0 +1,166 @@
+package com.mraof.minestuck.block;
+
+import com.mraof.minestuck.item.CassetteItem;
+import com.mraof.minestuck.tileentity.CassettePlayerTileEntity;
+import com.mraof.minestuck.util.CustomVoxelShape;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.item.ItemEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.state.BooleanProperty;
+import net.minecraft.state.EnumProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.stats.Stats;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityType;
+import net.minecraft.util.Hand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.Constants;
+
+import javax.annotation.Nullable;
+import java.util.function.Supplier;
+
+
+public class CassettePlayerBlock extends DecorBlock
+{
+	public static final BooleanProperty OPEN = BlockStateProperties.OPEN;
+	public static final EnumProperty<EnumCassetteType> CASSETTE = MSProperties.CASSETTE;
+	private final Supplier<TileEntityType<?>> tileType;
+	
+	public CassettePlayerBlock(Properties properties, CustomVoxelShape shape, Supplier<TileEntityType<?>> tileType)
+	{
+		super(properties, shape);
+		this.setDefaultState(this.stateContainer.getBaseState().with(CASSETTE, EnumCassetteType.NONE));
+		this.tileType = tileType;
+	}
+	
+	@Override
+	@SuppressWarnings("deprecation")
+	public boolean onBlockActivated(BlockState state, World worldIn, BlockPos pos, PlayerEntity player, Hand handIn, BlockRayTraceResult hit)
+	{
+		if(player.isSneaking())
+		{
+			state = state.cycle(OPEN);
+			TileEntity tileentity = worldIn.getTileEntity(pos);
+			ItemStack itemStack = ((CassettePlayerTileEntity) tileentity).getCassette();
+			worldIn.setBlockState(pos, state, 2);
+			if(tileentity instanceof CassettePlayerTileEntity && !state.get(OPEN))
+			{
+				worldIn.playEvent((PlayerEntity) null, Constants.WorldEvents.PLAY_RECORD_SOUND, pos, Item.getIdFromItem(itemStack.getItem()));
+				if(player != null)
+				{
+					player.addStat(Stats.PLAY_RECORD);
+				}
+			} else if(tileentity instanceof CassettePlayerTileEntity && state.get(OPEN))
+			{
+				worldIn.playEvent(1010, pos, 0);
```

Comment:
World event constants (in this case PLAY_RECORD_SOUND) is prefered over number ids here

Reply:
something like that?
`worldIn.playEvent((PlayerEntity) null, Constants.WorldEvents.PLAY_RECORD_SOUND, pos, 0);`

Reply:
Yep! (however you don't need the `(PlayerEntity) null` part)

Reply:
ok !